src/functions/functionsRouter.ts

```diff
@@ -0,0 +1,689 @@
+import {
+  ResponseListener,
+  SecretsManager,
+  SubscriptionManager,
+} from "@chainlink/functions-toolkit";
+import {
+  FunctionsResponse,
+  GatewayResponse,
+  RequestCommitment,
+  ThresholdPublicKey,
+} from "@chainlink/functions-toolkit/dist/types";
+import { BigNumber, BigNumberish, providers, Signer } from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { FunctionsSubscriptionDetails, Overrides } from "../shared/types";
+
+// DIRECT METHOD CALLS
+
+export const createSubscription = async (
+  hre: HardhatRuntimeEnvironment,
+  functionsRouterAddress: string,
+  consumerAddress?: string,
+  overrides?: Overrides
+): Promise<{ subscriptionId: BigNumber }> => {
+  const functionsRouter = await FunctionsSubscriptionManager.initialize({
+    hre,
+    functionsRouterAddress,
+    overrides,
+  });
+  return functionsRouter.createSubscription(consumerAddress);
+};
+
+export const fundSubscription = async (
+  hre: HardhatRuntimeEnvironment,
+  functionsRouterAddress: string,
+  linkTokenAddress: string,
```

Comment:
Does this user need to pass this in? can it be loaded from a registry?

Reply:
I tended to make functions signatures more general and not to tie it to the plugins registry. It is up to a user to pass it's own addresses or use the ones that are in the registry.